cmd/wio/commands/run/dependencies/target.go

```diff
@@ -0,0 +1,103 @@
+package dependencies
+
+import (
+    "hash/fnv"
+    "strconv"
+    "strings"
+    "wio/cmd/wio/toolchain/npm/semver"
+)
+
+type linkNode struct {
+    From     *Target
+    To       *Target
+    LinkInfo *TargetLinkInfo
+}
+
+type TargetLinkInfo struct {
+    Visibility string
+    Flags      []string
+}
+
+type Target struct {
+    Name                  string
+    Version               *semver.Version
+    Path                  string
+    Vendor                bool
+    HeaderOnly            bool
+    Flags                 []string
+    FlagsVisibility       string
+    Definitions           []string
+    DefinitionsVisibility string
+    hashValue             uint64
+}
+
+type targetSet struct {
+    tMap  map[uint64]*Target
+    links []*linkNode
+}
+
+// It creates 64-bit FNV-1a hash from name, version, flags, definitions, and headerOnly
+func (target *Target) hash() uint64 {
+    structStr := target.Name + SemverVersionToString(target.Version) + strings.Join(target.Flags, "") +
```

Comment:
I thought we could use hashmaps as keys?

Reply:
That does not work. He was using pointer as the key I realized after

Reply:
struct especially with slices do not have a hash function for them so go panics

Reply:
Oh lol I didn't realise that either. Well don't use an integer (like I said yesterday). Use a string at least. It's more robust that way because if two structs hash to the same integer, there's no way to tell them apart. But with string "hashes" there more guarantee that structs will be hashed uniquely

Reply:
strings could be very long so are you sure it won't cause any potential issue?